### PR TITLE
fix: unify session and automation API keys into a single credential with consistent header

### PR DESCRIPTION
**Pull request by malhotra5**
## Summary

Unifies authentication for the agent-server and automation backend so that **both backends use the same key value and the same `X-Session-API-Key` header**. Previously, the stack maintained two separate auth credentials with different header formats:

- Agent-server: `X-Session-API-Key: <key>` (via `VITE_SESSION_API_KEY`)
- Automation backend: `Authorization: Bearer <key>` (via a separate `VITE_AUTOMATION_API_KEY`)

After this PR, both backends authenticate via `X-Session-API-Key` using the single `VITE_SESSION_API_KEY` env var.

### Frontend (`src/api/automation-service/automation-service.api.ts`)
- The axios request interceptor now sets the `X-Session-API-Key` header (matching the agent-server pattern) instead of `Authorization: Bearer …`
- The key value is read from `VITE_SESSION_API_KEY` — the same env var used for agent-server auth
- `VITE_AUTOMATION_API_KEY` is removed entirely

### Dev scripts (`scripts/dev-with-automation.mjs`, `scripts/static-build.mjs`, `scripts/dev-static.mjs`)
- `localApiKey` is set equal to `sessionApiKey` — there is no separate automation key
- Removed the `AUTOMATION_LOCAL_API_KEY` env var override and the `OH_AUTOMATION_API_KEY_PATH` persistence path
- Removed the separate `automation-api-key.txt` persisted file (the session key file is the single source)
- Only `VITE_SESSION_API_KEY` is baked into static builds (no more `VITE_AUTOMATION_API_KEY`)
- Updated all comments to reflect the unified `X-Session-API-Key` header model

### Docker (`docker/entrypoint.sh`)
- Defaults `OPENHANDS_AUTOMATION_API_KEY`, `AUTOMATION_LOCAL_API_KEY`, and `AUTOMATION_AGENT_SERVER_API_KEY` to the session key value
- Individual override escape hatches are preserved for edge-case deployments

### Tests
- Updated test descriptions and assertions to reflect unified key model
- Removed tests for separate automation key persistence and override (`AUTOMATION_LOCAL_API_KEY`)
- Added assertion that `localApiKey === sessionApiKey`

## How auth works after this PR

| Target | Header | Key source |
|--------|--------|------------|
| Agent server | `X-Session-API-Key: <key>` | `VITE_SESSION_API_KEY` |
| Automation backend | `X-Session-API-Key: <key>` | `VITE_SESSION_API_KEY` |

**Single override**: Set `SESSION_API_KEY` (or `OH_SESSION_API_KEYS_0`) to control the key for both backends.

---
_This PR was updated by an AI agent (OpenHands) on behalf of the user._

---
**🐳 Docker images for this PR**

• **GHCR package:** https://github.com/OpenHands/agent-canvas/pkgs/container/agent-canvas

| Component | Value |
|---|---|
| **Image** | `ghcr.io/openhands/agent-canvas` |
| **Architectures** | amd64, arm64 |
| **Agent Server** | `ghcr.io/openhands/agent-server:1.23.1-python` |
| **Automation** | `openhands-automation==1.0.0a5` |
| **Commit** | `b157f716a3086f2d1068b6aa80fb3934ea7bae09` |

**Pull (multi-arch manifest)**
```bash
# Multi-arch manifest — Docker automatically pulls the correct architecture
docker pull ghcr.io/openhands/agent-canvas:sha-b157f71
```

**Run**
```bash
docker run -it --rm \
  -p 8000:8000 \
  ghcr.io/openhands/agent-canvas:sha-b157f71
```

**All tags pushed for this build**
```
ghcr.io/openhands/agent-canvas:sha-b157f71-amd64
ghcr.io/openhands/agent-canvas:fix-unify-api-keys-amd64
ghcr.io/openhands/agent-canvas:pr-681-amd64
ghcr.io/openhands/agent-canvas:sha-b157f71-arm64
ghcr.io/openhands/agent-canvas:fix-unify-api-keys-arm64
ghcr.io/openhands/agent-canvas:pr-681-arm64
ghcr.io/openhands/agent-canvas:sha-b157f71
ghcr.io/openhands/agent-canvas:fix-unify-api-keys
ghcr.io/openhands/agent-canvas:pr-681
```

**About Multi-Architecture Support**
- Each tag (e.g., `sha-b157f71`) is a **multi-arch manifest** supporting both **amd64** and **arm64**
- Docker automatically pulls the correct architecture for your platform
- Individual architecture tags (e.g., `sha-b157f71-amd64`) are also available if needed

### CODE REVIEW

File: scripts/dev-with-automation.mjs [hunk 1]
```diff
@@ -375,8 +365,8 @@ async function buildConfig(args, env = process.env) {
     // Data directories (same as dev-safe.mjs)
     stateDir,
 
-    // Auth
-    localApiKey,
+    // Auth — single key for both backends
+    localApiKey: sessionApiKey,
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/dev-with-automation.mjs]: Done — removed `localApiKey` from the config object entirely. All three call sites (L518, L619, L635) now use `config.sessionApiKey` directly, so the unified-key intent is obvious without tracing through an alias.

_This comment was generated by an AI agent (OpenHands) on behalf of the user._

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -76,6 +78,15 @@ if [ -z "${OH_SESSION_API_KEYS_0:-}" ] && [ -z "${SESSION_API_KEY:-}" ]; then
   export OH_SESSION_API_KEYS_0="$SESSION_API_KEY"
 fi
 
+# Both backends share the same API key value: the agent-server validates it
+# via `X-Session-API-Key` and the automation backend validates it via
+# `Authorization: Bearer …`.  Default OPENHANDS_AUTOMATION_API_KEY to the
+# session key so a single credential secures the whole stack.
+EFFECTIVE_SESSION_KEY="${OH_SESSION_API_KEYS_0:-${SESSION_API_KEY:-}}"
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply docker/entrypoint.sh:84]: Done — added a fail-fast guard. If `EFFECTIVE_SESSION_KEY` is empty the entrypoint now logs an error and exits with code 1 instead of silently exporting empty strings.

_This comment was generated by an AI agent (OpenHands) on behalf of the user._

File: scripts/dev-with-automation.mjs [hunk 3]
```diff
@@ -86,9 +86,8 @@ const DEFAULT_AUTOMATION_VERSION = SHARED_DEFAULTS.versions.automation;
 const DEFAULT_AUTOMATION_SDK_VERSION = SHARED_DEFAULTS.versions.automationSdk;
 const DEFAULT_BACKEND_PORT = SHARED_DEFAULTS.ports.agentServer;
 const DEFAULT_AUTOMATION_PORT = SHARED_DEFAULTS.ports.automation;
-// Where the auto-generated default automation API key is persisted. Static
-// frontend builds bake VITE_AUTOMATION_API_KEY at build time, so the default
-// must remain stable across restarts and --skip-build reuse.
+// Legacy path kept only so the export doesn't break any downstream importers.
+// The automation backend now reuses the session API key — no separate file.
 const DEFAULT_AUTOMATION_API_KEY_PATH = join(
   homedir(),
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/dev-with-automation.mjs]: Done — removed `DEFAULT_AUTOMATION_API_KEY_PATH` and its export entirely (no downstream consumers, and breaking changes are fine in beta).

_This comment was generated by an AI agent (OpenHands) on behalf of the user._

File: scripts/dev-with-automation.mjs [hunk 4]
```diff
@@ -652,7 +632,7 @@ function startAutomationBackend(config) {
           config.automationWorkspaceBase ||
           join(config.stateDir, "workspaces"),
         // Local API key for self-hosted auth (no cloud API needed)
-        AUTOMATION_LOCAL_API_KEY: config.localApiKey,
+        AUTOMATION_LOCAL_API_KEY: config.sessionApiKey,
```
[the remark this answers is not in the document]
Comment (malhotra5) [reply scripts/dev-with-automation.mjs:635]: Done — updated the comment to: `// Session API key for self-hosted auth — shared with agent-server via X-Session-API-Key header`.

_This comment was generated by an AI agent (OpenHands) on behalf of the user._
